### ISSUE TITLE
Linchpin validate fails on following Pinfile

### ISSUE THREAD

**Issue by samvarankashyap** [CentOS-PaaS-SIG/linchpin#778]
```
---
ember:
  topology:
    topology_name: ember
    resource_groups:
    - resource_group_name: libvirt-single
      resource_group_type: libvirt
      resource_definitions:
      - role: libvirt_node
        name: centos71
        uri: qemu:///system
        image_src: http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-1608.qcow2.xz
        count: 1
        networks:
          - name: default
  layout:
    inventory_layout:
      vars:
        hostname: __IP__
      hosts:
        example-node:
          count: 1
          host_groups:
            - example
```

Error:
```
(nummy) 🎩linchpin -vvv up
Processing target: ember
rundb_id: 2
uhash: e5816f
Topology 'OrderedDict([('topology_name', 'ember'), ('resource_groups', [OrderedDict([('resource_group_name', 'libvirt-single'), ('resource_group_type', 'libvirt'), ('resource_definitions', [OrderedDict([('role', 'libvirt_node'), ('name', 'centos71'), ('uri', 'qemu:///system'), ('image_src', 'http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-1608.qcow2.xz'), ('networks', [OrderedDict([('name', 'default')])]), ('count', 1)])])])])])' does not validate.  For more information run `linchpin validate`
(nummy) 🎩linchpin validate
Traceback (most recent call last):
  File "/home/srallaba/.virtualenvs/nummy/bin/linchpin", line 11, in <module>
    load_entry_point('linchpin', 'console_scripts', 'linchpin')()
  File "/home/srallaba/.virtualenvs/nummy/lib/python2.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/srallaba/.virtualenvs/nummy/lib/python2.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/srallaba/.virtualenvs/nummy/lib/python2.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/srallaba/.virtualenvs/nummy/lib/python2.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/srallaba/.virtualenvs/nummy/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/srallaba/.virtualenvs/nummy/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/srallaba/.virtualenvs/nummy/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/srallaba/workspace/lp_ws/nummy/linchpin/linchpin/shell/__init__.py", line 613, in validate
    "validate\n{3}".format(kind, target, outcome)
IndexError: tuple index out of range
```

**Comment by 14rcole** [comment]
@samvarankashyap did you fix this? I can't reproduce the error when I test against the most recent develop branch?

**Comment by samvarankashyap** [comment]
@14rcole I think its not fixed . Let me check it again .

**Comment by samvarankashyap** [comment]
@14rcole 
```
openshift-new:
  topology:
    topology_name: openshift
    resource_groups:
      - resource_group_name: test1
        resource_group_type: openshift
        resource_definitions:
          - name: testing
            role: openshift_inline
            kind: Namespace
          - name: anotesting
            role: openshift_inline
            kind: Namespace
        credentials:
          filename: open  # fetched from --creds-path is provided
          profile: default
```
linchpin validate says:
(k8s) 🎩linchpin validate
```
[OrderedDict([('topology_name', 'ember'), ('resource_groups', [OrderedDict([('resource_group_name', 'libvirt-single'), ('resource_group_type', 'libvirt'), ('resource_definitions', [OrderedDict([('role', 'libvirt_node'), ('name', 'centos71'), ('uri', 'qemu:///system'), ('image_src', 'http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-1608.qcow2.xz'), ('count', 1), ('networks', [OrderedDict([('name', 'default')])])])])])])])] not found in workspace
```
which should not !

**Comment by 14rcole** [comment]
@samvarankashyap so the reason that's happening there is that `topology_name` is under-indented.  linchpin therefore sees `topology: ` as a key that requires a value rather than a key with a child dict, and thinks that everything below is a file.

Unfortunately, linchpin only validates data inside of each `resource_definition`. We could expand validation or do some sort of multi-layer validation, but those are really our only options, and both will take a while to fix

**Comment by 14rcole** [comment]
Fixed by #892
